packages/app/src/debug/DebugContainer.cy.tsx

```diff
@@ -0,0 +1,161 @@
+    })
+
+    it('has no runs', () => {
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setUserFlag('isLoggedIn', true)
+      loginConnectStore.setProjectFlag('isProjectConnected', true)
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        render: (gqlVal) => {
+          return (
+            <DebugContainer
+              gql={gqlVal}
+            />
+          )
+        },
+      })
+
+      cy.findByTestId('debug-empty').contains(defaultMessages.debugPage.noRuns)
+    })
+  })
+
+  describe('render specs and tests', () => {
+    it('renders data when logged in and connected', () => {
+      const loginConnectStore = useLoginConnectStore()
+
+      loginConnectStore.setUserFlag('isLoggedIn', true)
+      loginConnectStore.setProjectFlag('isProjectConnected', true)
+      cy.mountFragment(DebugSpecsFragmentDoc, {
+        onResult: (result) => {
+          if (result.currentProject?.cloudProject?.__typename === 'CloudProject') {
+            const test = result.currentProject.cloudProject.runByNumber
+            const other = CloudRunStubs.failingWithTests as typeof test
+
+            result.currentProject.cloudProject.runByNumber = other
+          }
+        },
+        render: (gqlVal) => {
+          return (
+            <DebugContainer
+              gql={gqlVal}
+            />
+          )
+        },
+      })
+
+      // Only asserting that it is rendering the components for failed specs
+      cy.findByTestId('debug-header').should('be.visible')
+      cy.findByTestId('debug-spec-item').should('be.visible')
+    })
+  })
+
+  describe('testing util function: debugMapping', () => {
+    const mappingList = (specs: {id: string}[], tests: {specId: string, id: string}[]) => {
+      return specsList(specs, tests)
+    }
+
+    const createSpecs = (idArr: string[]) => {
+      const acc: {id: string}[] = []
+
+      idArr.forEach((val) => {
+        acc.push({ id: val })
+      })
+
+      return acc
+    }
+
+    it('maps correctly for a single spec', () => {
+      const spec = createSpecs(['a1c'])
+      const tests = [
+        { specId: 'a1c', id: 'random1' },
+        { specId: 'a1c', id: 'random2' },
+      ]
+
+      const debugMappingArray = mappingList(spec, tests)
+
+      expect(debugMappingArray).to.have.length(1)
+      expect(debugMappingArray[0]).to.deep.equal({ spec: { id: 'a1c' }, tests: [{ specId: 'a1c', id: 'random1' }, { specId: 'a1c', id: 'random2' }] })
+    })
+
+    it('maps correctly for multiple specs and test', () => {
+      const specs = createSpecs(['123', '456', '789'])
+      const tests = [
+        { specId: '123', id: 'random1' },
+        { specId: '456', id: 'random2' },
+        { specId: '456', id: 'random3' },
+        { specId: '789', id: 'random4' },
+        { specId: '123', id: 'random6' },
+      ]
+
+      const debugMappingArray = mappingList(specs, tests)
+
+      expect(debugMappingArray).to.have.length(3)
```

Comment:
Unit testing with Cypress is quite a painful - we should definitely look to get [Vitest](https://vitest.dev/) in (using Vite, like for the frontend - so using the same config - but a very fast, terminal based runner, similar to Jest - which is perfect for these kind of tests).